homeassistant/components/mqtt/__init__.py

```diff
@@ -198,6 +215,20 @@ def valid_publish_topic(value: Any) -> str:
                  default=DEFAULT_PAYLOAD_NOT_AVAILABLE): cv.string,
 })
 
+MQTT_ENTITY_DEVICE_INFO_SCHEMA = vol.All(vol.Schema({
+    vol.Optional(CONF_IDENTIFIERS, default=list):
+        vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_CONNECTIONS, default=list):
+        vol.All(cv.ensure_list, [vol.Schema({
+            vol.Required(CONF_TYPE): vol.Any(*device_registry.CONNECTION_KEYS),
+            vol.Required(CONF_IDENTIFIER): cv.string,
```

Comment:
I don't really like the name of this option (`identifier`) in this context. Its name (base off of https://developers.home-assistant.io/docs/en/device_registry_index.html) is very close to the `identifiers` option but it has a totally different format. Should this be renamed to `value` or should it rather stick to the dev docs' `identifier`?

Reply:
Why is this a dict to begin with?

Reply:
Well primarily to make it more consistent with the device registry naming. One option would be:

```json
{
  ...
  "device": {
    "connections": {
        "mac": "F7:19:18:29:93:1A",
        "other": "connection_id"
    }
  }
}
```

but that would for example not support devices with multiple MAC addresses (probably doesn't come up too often, but I think it's good to support it from the start to prevent breaking changes).

The other option would be:

```json
{
  ...
  "device": {
    "connections": [{
        "mac": "F7:19:18:29:93:1A"
    }, {
        "mac": "E8:2A:29:3A:A4:2B"
    }]
  }
}
```

Reply:
I like the last one. That is also how we store it internally.

Reply:
Ok, updated to that.